### PR TITLE
MODINREACH-201 Adapt mod-inn-reach to the request schema changes 

### PR DESCRIPTION
**Pull request by bisecomsergiy**
### Purpose
Adapt the module to new request schema

### Approach

- Updated RequestDTO by new properties;
- Added new properties to RequestServiceimpl;
- Indicated new circulation module version in ModuleDescriptor; 

### Learning
https://issues.folio.org/browse/MODINREACH-201

### CODE REVIEW

File: src/main/java/org/folio/innreach/domain/service/ItemService.java [hunk 1]
```diff
@@ -13,4 +13,6 @@
 
   Optional<InventoryItemDTO> findItemByBarcode(String barcode);
 
+  Optional<InventoryItemDTO> find(UUID itemId);
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/ItemService.java]: the method is inherited from `BasicService`, no need to define it

File: src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java [hunk 2]
```diff
@@ -289,9 +291,14 @@ public CheckOutResponseDTO renewLoan(RenewLoanRequestDTO renewLoan) {
   }
 
   private void cancelRequest(RequestDTO request, String reason) {
+    var item = itemService.find(request.getItemId()).orElse(null);
+    var holding = item == null ? null : holdingsService.find(item.getHoldingsRecordId()).orElse(null);
+
     request.setStatus(RequestStatus.CLOSED_CANCELLED);
     request.setCancellationReasonId(INN_REACH_CANCELLATION_REASON_ID);
     request.setCancellationAdditionalInformation(reason);
+    request.setInstanceId(holding == null ? null : holding.getInstanceId());
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java]: I guess the `instanceId` should be already set as we load the request from circulation API
https://github.com/folio-org/mod-inn-reach/blob/91f709c443c9e108eb5d270ce79045dae351e939/src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java#L210-L211

File: src/main/java/org/folio/innreach/domain/service/ItemService.java (hunk 1 above)
Comment (bisecomsergiy) [reply src/main/java/org/folio/innreach/domain/service/ItemService.java]: It is moved, thank you

File: src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java (hunk 2 above)
Comment (bisecomsergiy) [reply src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java]: The call for holding is moved and property set from request. Thank you.

File: src/main/java/org/folio/innreach/domain/dto/folio/circulation/RequestDTO.java [hunk 3]
```diff
@@ -17,9 +17,11 @@
 @Builder
 public class RequestDTO {
   private UUID id;
+  private UUID instanceId;
   private UUID itemId;
   private RequestStatus status;
   private String requestType;
+  private String requestLevel;
```
Comment (alexanderkurash) [inline src/main/java/org/folio/innreach/domain/dto/folio/circulation/RequestDTO.java:25]: String -> enum

File: src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java [hunk 4]
```diff
@@ -158,6 +158,8 @@ public void createItemRequest(InnReachTransaction transaction, Holding holding,
     //creating and sending new request
     var newRequest = RequestDTO.builder()
       .requestType(requestType.getName())
+      .requestLevel("Item")
+      .instanceId(holding == null ? null : holding.getInstanceId())
       .itemId(item.getId())
```
Comment (alexanderkurash) [inline src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java]: Add holdingsRecordId

File: src/main/java/org/folio/innreach/domain/dto/folio/circulation/RequestDTO.java (hunk 3 above)
Comment (bisecomsergiy) [reply src/main/java/org/folio/innreach/domain/dto/folio/circulation/RequestDTO.java:25]: Used enum, thank you

File: src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java (hunk 4 above)
Comment (bisecomsergiy) [reply src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java]: Added necessary property. Thank you.

File: src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java [hunk 5]
```diff
@@ -264,11 +265,16 @@ private void cancelTransaction(InnReachTransaction transaction) {
   @Override
   public void createRecallRequest(UUID recallUserId, UUID itemId) {
     var pickupServicePoint = getDefaultServicePointIdForPatron(recallUserId);
+    var item = itemService.getItemByHrId(itemId.toString());
+    var holding = item == null ? null : holdingsService.find(item.getHoldingsRecordId()).orElse(null);
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java]: We store identifiers of instance, holding and item in `TransactionHold` when initial request is successfully created. So, at this point there is no need to load the records from inventory to get them, you can relay on these values `folioItemId`, `folioHoldingId`, `folioInstanceId`

Comment (bisecomsergiy) [reply src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java]: It is done, thank you!

**Review by oliinyko — APPROVED** [review]

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**